joss/paper.md

```diff
@@ -0,0 +1,103 @@
+- id (e.g., `Id., at 101`)
+
+It also offers tools for efficiently aggregating like citations and annotating
+citations with custom markup.
+
+# Statement of need
+
+Citations are the bedrock of legal writing and a frequent topic of legal
+research, but few open-source tools exist for extracting them from legal
+texts. Because of this, researchers have historically relied on proprietary
+citation data provided by vendors like LexisNexis and Westlaw [e.g., @Black2013;
+@Fowler2007; @SpriggsII2000] or have used their own personal scripts to parse
+such data from texts ad hoc [e.g., @Clark2012; @Fowler2008]. By providing an
+open-source, standardized alternative to these approaches, `eyecite` promises
+to increase scholarly transparency and consistency. It also promises to give
+researchers the extendability and flexibility to develop new methods of
+citation analysis that are currently not possible under the prevailing
+approaches.
+
+For example, one burgeoning research agenda seeks to apply machine learning
+techniques to citation analysis, either to recommend relevant authorities to
+legal practitioners [@HoForthcoming], model the topography of the legal search
+space [@Dadgostari2021; @Leibon2018], or automatically detect and label
+the semantic purpose of citations in texts [@Sadeghian2018]. One obvious
+application of `eyecite` would be to use it to generate empirical training
+data for these kinds of machine learning tasks.
+
+To facilitate those kinds of projects and more, `eyecite` exposes significant
+entity metadata to the user. For case citations, `eyecite` parses and
+exposes information regarding a citation's textual position, year, normalized
+reporter, normalized court, volume, page, pincite page, and accompanying
+parenthetical text, as well as `eyecite`'s best guess at the names of the
+plaintiff and defendant of the cited case.  For statutory citations, `eyecite`
+parses and exposes information regarding a citation's textual position, year,
+normalized reporter, chapter, section, publisher, and accompanying parenthetical
+text. It does this using a
+[database](https://github.com/freelawproject/reporters-db) of thousands of
+regular expressions that have been accumulated and refined over the course of
+years of research.
+
+Recognizing that in addition to obtaining comprehensive results, researchers
+often want to parse many documents and citations at once, `eyecite` is also
+designed with an eye toward performance: it makes use of the Hyperscan library
+[@Wang2019] to tokenize and parse its input text in a highly efficient fashion.
+Additionally, `eyecite` can heuristically resolve short case, supra, and id
+citations to their appropriate full case antecedents, and it integrates well
+with custom resolution logic. Finally, for practical applications, it can
+"annotate" found citations with custom markup (like HTML links)  and re-insert
+that markup into the appropriate place in the original text.
+
```

Comment:
Should we mention that the original text can be very different from the text used for extraction? Something like "To allow for annotation of rich text, `eyecite` uses the diff-match-patch library to insert annotations into original text even if the text requires cleaning prior to extraction."